core/src/apps/eos/get_public_key.py

```diff
@@ -0,0 +1,42 @@
+from trezor import wire
+from trezor.crypto import base58
+from trezor.crypto.curve import secp256k1
+from trezor.crypto.hashlib import ripemd160
+from trezor.messages.EosGetPublicKey import EosGetPublicKey
+from trezor.messages.EosPublicKey import EosPublicKey
+
+from apps.common import paths
+from apps.eos.helpers import validate_full_path
+from apps.eos.layout import require_get_public_key
+
+
+def _ripemd160_32(data: bytes) -> bytes:
+    return ripemd160(data).digest()[:4]
+
+
+def _public_key_to_wif(pub_key: bytes) -> str:
+    if len(pub_key) == 65:
+        head = 0x03 if pub_key[64] & 0x01 else 0x02
+        compresed_pub_key = bytes([head]) + pub_key[1:33]
+    elif len(pub_key) == 33:
+        compresed_pub_key = pub_key
+    else:
+        raise wire.DataError("invalid public key length")
+    return "EOS" + base58.encode_check(compresed_pub_key, _ripemd160_32)
+
+
+def _get_public_key(node):
+    seckey = node.private_key()
+    public_key = secp256k1.publickey(seckey, True)
+    wif = _public_key_to_wif(public_key)
+    return wif, public_key
+
+
+async def get_public_key(ctx, msg: EosGetPublicKey, keychain):
+    await paths.validate_path(ctx, validate_full_path, path=msg.address_n)
```

Comment:
we have changed this again in the meantime. please update this call, see this commit https://github.com/trezor/trezor-firmware/commit/8aa60e6cfd84ba35613e5eed0088fb43e97aa8e0 for details

Reply:
Done